app/src/main/java/com/example/habitsmasher/ui/history/AddHabitEventDialog.java

```diff
@@ -0,0 +1,237 @@
+
+        // Add listener to date text to open date picker
+        _eventDateText.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                openDatePickerDialog();
+            }
+        });
+
+        // Add listener to cancel button which closes dialog
+        _cancelNewEvent.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Cancel");
+                getDialog().dismiss();
+            }
+        });
+
+        // Add listener to confirm button that adds events to database and closed dialog
+        _confirmNewEvent.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.O)
+            @Override
+            public void onClick(View view) {
+                // Check if event data is valid
+                if (checkNewHabitEventIsValid()) {
+                    // If everything is valid, add event to database, events list, and close dialog
+                    HabitEvent newEvent = new HabitEvent(_eventDate, _eventCommentInput, _selectedImage);
+                    _habitEventListFragment.addNewHabitEvent(newEvent);
+                    _habitEventListFragment.addHabitEventToDatabase(newEvent.getStartDate(), newEvent.getComment(), newEvent.getId(), newEvent.getPictureUri(), _username);
+                    getDialog().dismiss();
+                }
+
+            }
+        });
+
+        // Add listener to image view
+        _eventPictureView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // Open gallery to let user pick photo
+                Intent pickPhoto = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+                startActivityForResult(pickPhoto, 1);
+            }
+        });
+
+        return view;
+    }
+
+
+    /**
+     * Reference: https://stackoverflow.com/questions/10165302/dialog-to-pick-image-from-gallery-or-from-camera
+     * Override onActivityResult to handle when user has selected image
+     * @param requestCode
+     * @param resultCode
+     * @param imageReturnedIntent
+     */
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent imageReturnedIntent) {
+        super.onActivityResult(requestCode, resultCode, imageReturnedIntent);
+        switch(requestCode) {
+            case 0:
+            case 1:
+                if (resultCode == RESULT_OK) {
+                    // Set selected picture
+                    _selectedImage = imageReturnedIntent.getData();
+                    _eventPictureView.setImageURI(_selectedImage);
+                }
+                break;
+        }
+
+    }
+
+    /**
+     * Opens date picker
+     */
+    private void openDatePickerDialog(){
```

Comment:
Also, a lot of the **date** stuff has been extracted out by @JacobNguyen71 as well. I think it would be good to touch base with him, or create a task to extract the date picker stuff to use the new extracted class **later** (still this sprint).

Reply:
Two avoid blocking, I'll make a task to do it once that has been merged into develop

Reply:
Cool, I'd just make the task now and not wait until this is merged (because we will both likely forget)